### PR TITLE
Set of changes to prepare tester.py to move provision code out 

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/K9BLCYcp/688-refactor-separate-provision-logic-from-test-logic

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/sct_config.py [hunk 1]
```diff
@@ -1680,3 +1680,12 @@ def dump_help_config_yaml(self):
             ret += "{help_text}{name}: {default}\n\n".format(help_text=help_text, default=default, **opt)
 
         return ret
+
+
+def init_sct_config_from_environment() -> SCTConfiguration:
```
Comment (bentsi) [inline sdcm/sct_config.py:1685]: better name will be `init_and_verify_sct_config`

File: sdcm/utils/setup.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+from sdcm.utils.decorators import retrying
+from sdcm.utils.docker_utils import ContainerManager
+from sdcm.utils.get_username import get_username
+from sdcm.utils.ldap import LdapServerNotReady
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Setup:
```
Comment (bentsi) [inline sdcm/utils/setup.py:22]: TestConfig

**Comment by bentsi** [comment]
let's run PR provision on GCE first, than on AWS

File: sdcm/sct_config.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/sct_config.py]: done

File: sdcm/utils/setup.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/utils/setup.py]: done

**Review by bentsi — APPROVED** [review]

**Pull request merged**